### DISCUSSION TITLE
Route and pretty urls beetween private and public folder structure in pages

### DISCUSSION THREAD

**Discussion by gianfabiopezzolla** [vercel/next.js#28599] [Help]
Hi there,
I'm developing an app that include some public pages and other destinated to logged user.

So my folder structure is like:

pages
-- private
---- privatePages
-- public
---- publicPages

I'm trying to remove "public" and "private" from the url, but with no luck. I've tried to use "as" in <Link> and it seems to apparently work, but when I refresh the page, the 404 page is shown.

How can I handle with this?

thanks a lot

**Comment by delbaoliveira** [comment]
Hi @SAGOlab 👋🏼 

Will the private pages contain dynamic information or will the information be the same for all users? 

Without knowing too much about your project, you may want to consider [Dynamic Routes](https://nextjs.org/docs/routing/dynamic-routes) and/or [Rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) (to map an incoming request path to a different destination).
